packages/drivers/odsp-driver/src/odspDriverUrlResolver2.ts

```diff
@@ -62,12 +62,13 @@ export class OdspDriverUrlResolver2 implements IUrlResolver {
      */
     public async resolve(request: IRequest): Promise<IOdspResolvedUrl> {
         const requestToBeResolved = { headers: request.headers, url: request.url };
-        let sharingLink: string | undefined;
+        let sharingLinkP: Promise<string> | undefined;
+        const isSharingLink = requestToBeResolved.headers?.[SharingLinkHeader.isSharingLink];
         try {
             const url = new URL(request.url);
             // Check if the url is the sharing link.
-            if (request.headers?.[SharingLinkHeader.isSharingLink]) {
-                sharingLink = request.url.split("?")[0];
+            if (isSharingLink) {
```

Comment:
hmm, I wonder if Teams would ever want to send in a sharing link promise instead of a sharing link directly... could maybe be better for Teams performance if we just take in a sharing link promise in the header and treat everything the same from then on.

Reply:
I think we can take this as a future improvement because right now they generate the sharing link only.